blaze-client/src/main/scala/org/http4s/blaze/client/PoolManager.scala

```diff
@@ -90,7 +93,7 @@ private final class PoolManager[F[_], A <: Connection[F]](
 
   private var isClosed = false
   private val curTotal = new AtomicInteger(0)
-  private val allocated = mutable.Map.empty[RequestKey, Int]
+  private val allocated = new ConcurrentHashMap[RequestKey, Int]()
   private val idleQueues = mutable.Map.empty[RequestKey, mutable.Queue[PooledConnection]]
   private var waitQueue = mutable.Queue.empty[Waiting]
```

Comment:
I mean, if we're worried about thread safety of this code in general, then I'd guess we should protect these two guys too.